### DISCUSSION TITLE
Multiple Embedded Boundaries

### DISCUSSION THREAD

**Discussion by connor-powers** [BLAST-WarpX/warpx#5400] [Q&A] [answered]
Hello! I'm still in the process of learning how to use WarpX, and I was wondering if there's any way to create multiple embedded boundaries? I'm looking to set up an RZ simulation with EBs on the lower and upper radial boundaries, and I've set up the lower one but I'm having trouble determining how I might also create the upper one with a different geometry and potential function. Thanks!

**Comment by RemiLehe** [comment, marked as answer]
Hi,

Thanks for your question.
You can create multiple embedded boundaries with a single implicit function, keeping in mind that WarpX will place an embedded boundary wherever that function switches from positive to negative values. So can create several, disjoint embdedded boundaries this way.

As an example, you can create to cylinders having different radii with:
```
warpx.eb_implicit_function = "(x-rmin)*(x-rmax)"
```
(`x` is interpreted as `r` in RZ geometry)

This works because the implicit function will be negative inbetween `rmin` and `rmax` and positive outside.

Does this answer your question?

**Reply by connor-powers** [reply]
Great! That does answer it, thank you.

**Comment by ncast-dev** [comment]
**DIGI-337 — QAMagicMockFilter ✅**

`shared/log_filters.py` existe déjà avec `QAMagicMockFilter` et `shared/bot_logging.py` applique le filtre sur le `StreamHandler`.

5/5 tests GREEN — tree propre, pas de nouveau commit nécessaire.

Hash dernier commit: `ed8ab1b`
